### PR TITLE
Fix bugs when listing data and verifying users authorization

### PR DESCRIPTION
**Pull request by netoax**
What does this implement/fix? Explain your changes.
---------------------------------------------------
This patch introduces a lot of relevant bug fixes, the main are described below:

- Fix list data correctly by `deviceId` and `sensorId`;
- Refactor data controller in order to improve readability;
- Enable fine-grained authorization so that the user can't access other users' data.

Does this close any currently open issues?
------------------------------------------
Nops.

Any relevant logs, error output, etc?
-------------------------------------
Nops.

Any other comments?
-------------------
Nops.

Where has this been tested?
---------------------------
Nops.

**Operating System/Platform:** macOS

**NPM Version:** 6.14.4

**NodeJS Version:** v12.16.2

### CODE REVIEW

**Review by lcbm — CHANGES_REQUESTED** [review]
Awesome patch :100: 

A few notes:

- There are unused packages in `go.mod`, could you run `go mod tidy`? :grimacing: 
- One of the commit messages have extra indentation, could you fix it? :stuck_out_tongue: 
- I noticed that `go vet`, during the `pre-commit-hook` doesn't work as expected. It should run `go vet ./...` but it runs `go vet .` instead -- the `go vet` command will not find the package and always fail. Would you kindly add a commit removing it? We can take a better look at this later :smile:

File: pkg/data/data_store.go [hunk 1]
```diff
@@ -12,7 +10,7 @@ import (
 const collection = "data"
 
 type IDataStore interface {
-	Get(order string, skip, take int, startDate, finishDate time.Time)
+	Get(query *entities.Query)
 	Save(data entities.Data)
 	Delete(deviceID string)
 }
```
Comment (lcbm) [inline pkg/data/data_store.go]: Shouldn't the interfaces have the return types too? :point_down: 

```Go
type IDataStore interface {
	Get(query *entities.Query) ([]entities.Data, error)
	Save(data entities.Data) error
	Delete(deviceID string) error
}
```

**Review by lcbm — APPROVED** [review]
I made a suggestion to standardize interface implementations, but since it doesn't affect code functionality, and everything else is great, I'm approving this  :confetti_ball:

File: pkg/data/data_store.go (hunk 1 above)
Comment (netoax) [reply pkg/data/data_store.go]: Done

**Pull request merged**